File: labrabota8-gr12-MoonLory/Example8.2.1/app/src/main/java/com/example/example821/MainActivity.java
package com.example.example821;

import androidx.appcompat.app.AppCompatActivity;

import android.gesture.Gesture;
import android.gesture.GestureLibraries;
import android.gesture.GestureLibrary;
import android.gesture.GestureOverlayView;
import android.gesture.Prediction;
import android.os.Bundle;
import android.view.Menu;
import android.widget.TextView;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity implements GestureOverlayView.OnGesturePerformedListener {

    GestureLibrary gLib;
    GestureOverlayView gestures;

    TextView tvOut;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        gLib = GestureLibraries.fromRawResource(this, R.raw.gesture);
        if (!gLib.load()) {
            finish();
        }

        gestures = (GestureOverlayView) findViewById(R.id.gestureOverlayView1);
        gestures.addOnGesturePerformedListener(this);

        tvOut = findViewById(R.id.textView1);
    }

    @Override
    public void onGesturePerformed(GestureOverlayView overlay, Gesture gesture) {
        //Создаёт ArrayList c загруженными из gestures жестами
        ArrayList<Prediction> predictions = gLib.recognize(gesture);

        if (predictions.size() > 0) {
            Prediction prediction = predictions.get(0);

            if (prediction.score > 1.0) {
                if (prediction.name.equals("one")) {
                    tvOut.setText("1");
                } else if (prediction.name.equals("Stop")) {
                    tvOut.setText("stop");
                } else if (prediction.name.equals("two")) {
                    tvOut.setText("2");
                }
            } else {
                tvOut.setText("Жест неизвестен");
            }
        }
    }
}

File: labrabota9-gr12-MoonLory/Lab9_3_1/app/src/main/java/com/example/lab9_part1/MainActivity.kt
package com.example.lab9_part1

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity

import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    val PREFS_GAME = "com.example.lab9_3_1.GamePlay"

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        btnLogin.setOnClickListener{
            val sharedPref = getSharedPreferences(PREFS_GAME, MODE_PRIVATE)
            val editor = sharedPref.edit()

            editor.putString("NAME", txtName.text.toString())
            editor.apply()

            val intent = Intent(this, PlayActivity::class.java)
            startActivity(intent)
        }
    }
}